### DISCUSSION TITLE
Timeout when adding a device

### DISCUSSION THREAD

**Discussion by dapasternak** [edgexfoundry/community#253] [Q&A]
Hi, I am new so sorry if this is an obvious one

I am following the tutorial https://docs.edgexfoundry.org/3.1/walk-through/Ch-WalkthroughProvision/ and have been able to add the Device profile and service but when I try to add a device using the following CURL, I am getting a timeout but I do not see any other information

curl -X 'POST' 'http://localhost:59881/api/v3/device' -d '[{"apiVersion" : "v3", "device": {"name": "countcamera1","description": "human and dog counting camera #1","adminState": "UNLOCKED","operatingState": "UP","labels": ["camera","counter"],"location": "{lat:45.45,long:47.80}","serviceName": "camera-control-device-service","profileName": "camera-monitor-profile","protocols": {"camera-protocol": {"camera-address": "localhost","port": "1234","unitID": "1"}}}}]'

Response
<html><head><title>Timeout</title></head><body><h1>Timeout</h1></body></html>

Any ideas? I read somewhere that this could be an issue with v3?

**Comment by lenny-goodell** [comment]
Typically this means the service that is register `camera-control-device-service` is not responding to the request to validate and add the device. 
Please check the logs for Core Metadata to get more details.

Is `camera-control-device-service` your custom device service? 
You have obviously registered it, otherwise you would get a different error.

**Reply by dapasternak** [reply]
That is correct Lenny, it is the custom service as described in the tutorial 
![image](https://github.com/edgexfoundry/community/assets/160193200/145d835f-431f-4bdf-aa3c-2771d3373e4e)
https://docs.edgexfoundry.org/3.1/walk-through/Ch-WalkthroughDeviceService/

I will try to check the logs for core metadata

**Reply by lenny-goodell** [reply]
@dapasternak , that walk-thru needs to be updated for V3. The issue is there isn't an actual `camera-control-device-service` running to accept the validate/add request from Core Metadata when you submit the `AddDevice` request.

Try changing the to use `device-virtual` in you `AddDevice` request.

You can also use the UI to add devices, which is easier as it nows the device service specific elements needed and will prompt you for them.

![image](https://github.com/edgexfoundry/community/assets/44779287/838edc03-cfe3-43a3-add9-e04ed9145bdc)
![image](https://github.com/edgexfoundry/community/assets/44779287/45e3b499-5d21-4eec-931f-42c43f5997e5)
![image](https://github.com/edgexfoundry/community/assets/44779287/f8f910c6-5016-4fcf-8c27-c108bb772737)
![image](https://github.com/edgexfoundry/community/assets/44779287/d1bc5f88-6b9a-4ec6-bb13-80d38cfcd4cc)

**Reply by lenny-goodell** [reply]
I have submitted a bug against the documentation for this issue:
https://github.com/edgexfoundry/edgex-docs/issues/1338

**Reply by dapasternak** [reply]
@lenny-intel 
Thank you, Lenny, I will try as suggested and keep you posted.

Just for my understanding, what you are implying is that though I am trying to add a service "interface" with
![image](https://github.com/edgexfoundry/community/assets/160193200/542c035f-3fec-44df-b649-476a5e3a3b1e)

The actual service does not exist at the base address mentioned "baseAddress": "camera-device-service:59990"right?

**Reply by lenny-goodell** [reply]
Correct.

**Reply by dapasternak** [reply]
On that response @lenny-intel, can I build that service if it exists on GitHub and deploy it in EdgeX?

**Reply by lenny-goodell** [reply]
It doesn't exist. For this example, any device service will work and they all self-register.
BTW, you don't need to build the service. There are release docker images for all the services. Device REST and Device Virtual are is the default reference compose file, which it looks like you are running.
